pkg/service/controller.go

```diff
@@ -271,7 +271,16 @@ func (c *ControllerService) ControllerPublishVolume(
 
 	err = c.virtClient.AddVolumeToVM(c.infraClusterNamespace, vmName, addVolumeOptions)
 	if err != nil {
-		klog.Errorf("Failed adding volume %s to VM %s, %v", dvName, vmName, err)
+		// Need to ignore already exists error due to multiple add requests
+		if !strings.Contains(err.Error(), fmt.Sprintf("Unable to add volume [%s] because it already exists", dvName)) {
+			klog.Errorf("failed adding volume %s to VM %s, %v", dvName, vmName, err)
+			return nil, err
+		}
```

Comment:
exact error messages are not a versioned part of our api. This message could change. 

To avoid this error, we probably need to perform a VM get here, see if the volume exists, and only attempt to add it if it does not exist.

Reply:
The problem with that approach is that it is racy. And it is still possible to get that message, but since it will get retried, it will pass next go around.